res/utils.py

```diff
@@ -248,6 +252,42 @@ def json_datetime_converter(json_text):
 
     return json.dumps(json_text, default = datetime_converter)      
 
+def create_session(account_id, role):
+    """Create boto session using the IAM Role or the profile name
+
+    Args:
+        account_id (str): the AWS account id
+        role (str): The IAM role name or the profile name
+
+    Returns:
+        botocore.session: boto3 session
+    """
+    try:
+        session = boto3.Session(profile_name=role) 
```

Comment:
are you sure the cached sessions are reusable from the profile name ? is not `AssumeRoleSession{account_id}` instead ?

Reply:
it just to preserve the capability to use both profile and assume role feature.
the cached session is here used in the except section and for the assumerole only.
we can remove the session creation through profile from code if you want but keeping it can be useful when we will create the MR request to Janiko repository.